client/my-sites/posts/post-controls.jsx

```diff
@@ -1,251 +1,228 @@
+		} );
+	}
+
+	if ( userCan( 'delete_post', post ) ) {
+		if ( 'trash' === post.status ) {
+			controls.main.push( {
+				className: 'trash is-scary',
+				icon: 'trash',
+				onClick: onDelete,
+				text: translate( 'Delete Permanently' ),
 			} );
-
-			mainControls = this.buildControls( availableControls );
-			moreControls = this.buildControls( extraControls );
-			postControlsClass += ( extraControls.length <= 2 ) ? ' post-controls--desk-nomore' : '';
-
-			return (
-				<div className={ postControlsClass }>
-					<ul className="post-controls__pane post-controls__more-options">
-						{ moreControls }
-					</ul>
-					<ul className="post-controls__pane post-controls__main-options">
-						{ mainControls }
-					</ul>
-				</div>
-			);
 		} else {
-			mainControls = this.buildControls( availableControls );
-
-			return (
-				<div className={ postControlsClass }>
-					<ul className="post-controls__pane post-controls__main-options">
-						{ mainControls }
-					</ul>
-				</div>
-			);
+			controls.main.push( {
+				className: 'trash',
+				icon: 'trash',
+				onClick: onTrash,
+				text: translate( 'Trash' ),
+			} );
 		}
-	},
-
-} );
-
-export default connect(
-	( state, props ) => ( {
-		isPublicizeEnabled: isPublicizeEnabled( state, props.site.ID, props.post.type ),
-	} )
-)( PostControls );
+	}
+
+	if ( ( 'publish' === post.status || 'private' === post.status ) && userCan( 'edit_post', post ) ) {
+		controls.main.push( {
+			className: 'copy',
+			href: `/post/${ site.slug }?copy=${ post.ID }`,
+			icon: 'clipboard',
+			onClick: copy,
+			text: translate( 'Copy' ),
+		} );
+	}
+
+	// More Controls (behind ... more link)
+	if ( controls.main.length > 4 && fullWidth ) {
+		controls.more = controls.main.splice( 4 );
+	} else if ( controls.main.length > 2 && ! fullWidth ) {
+		controls.more = controls.main.splice( 2 );
+	}
+
+	if ( controls.more.length ) {
+		controls.main.push( {
+			className: 'more',
+			icon: 'ellipsis',
+			onClick: onShowMore,
+			text: translate( 'More' ),
+		} );
+
+		controls.more.push( {
+			className: 'back',
+			icon: 'chevron-left',
+			onClick: onHideMore,
+			text: translate( 'Back' ),
+		} );
+	}
+
+	return controls;
+};
+
+const getControlElements = controls => controls.map( ( control, index ) =>
+	<li
+		className={ classNames( { 'post-controls__disabled': control.disabled } ) }
+		key={ index }
+	>
+		<a
+			className={ `post-controls__${ control.className }` }
+			href={ control.href }
+			onClick={ control.disabled ? noop : control.onClick }
+			target={ control.target ? control.target : null }
+		>
+			<Gridicon icon={ control.icon } size={ 18 } />
+			<span>
+				{ control.text }
+			</span>
+		</a>
+	</li>
+);
+
+export const PostControls = props => {
+	const { main, more } = getAvailableControls( props );
+	const classes = classNames(
+		'post-controls',
+		{ 'post-controls--desk-nomore': more <= 2 }
+	);
+
+	return (
+		<div className={ classes }>
+			{ more.length &&
```

Comment:
This condition is causing `0` to be output when there are no "More" items. You can see this on the All Sites Drafts screen:

![image](https://cloud.githubusercontent.com/assets/1779930/22162923/c7b7d9b6-df1e-11e6-9c95-ebd872b3b474.png)

https://wordpress.com/posts/my/drafts

The way the condition reads, since the first part is not truthy, it uses the value on the left hand of the `&&`, and React will happily output `0`. You need this to either evaluate to a boolean, `null`, or `undefined`.

I suggest changing to `more.length > 0 &&`